tests.py

```diff
@@ -436,40 +635,384 @@ def __getitem__(self, key):
+                    'user': {
+                        'id': self.user_id,
+                        'name': 'user_name changed',
+                        'disable_toopher_auth': True
+                    }
+                })
+            )
+        })
+        pairing.refresh_from_server(self.api)
+        self.assertEqual(self.api.advanced.raw.client.last_called_method, 'GET')
+        self.assertEqual(pairing.id, self.id)
+        self.assertEqual(pairing.user.id, self.user_id)
+        self.assertEqual(pairing.user.name, 'user_name changed')
+        self.assertTrue(pairing.user.disable_toopher_auth)
+        self.assertFalse(pairing.enabled)
+        self.assertFalse(pairing.pending)
+
+    def test_get_qr_code_image(self):
+        response = {'id': 'id',
+                    'enabled': True,
+                    'pending': True,
+                    'user': self.user }
+        pairing = toopher.Pairing(response)
+
+        with open('qr_image.png', 'rb') as qr_image:
+            self.api.advanced.raw.client = HttpClientMock({
+                'qr/pairings/{0}'.format(pairing.id): (200,
+                                                       qr_image.read()
+                )
+            })
+            qr_image_data = pairing.get_qr_code_image(self.api)
+            self.assertEqual(self.api.advanced.raw.client.last_called_method, 'GET')
+            with open('new_image.png', 'wb') as new_image:
+                new_image.write(qr_image_data)
+
+            im = Image.open('new_image.png')
+            try:
+                im.verify()
+            except:
+                self.assertTrue(False)
+            os.remove('new_image.png')
+
+    def test_get_reset_link(self):
+        response = {'id':self.id,
+                    'enabled': False,
+                    'pending': True,
+                    'user': self.user }
+        pairing = toopher.Pairing(response)
+
+        self.api.advanced.raw.client = HttpClientMock({
+            'pairings/{0}/generate_reset_link'.format(self.id): (200,
+                json.dumps({
+                    'url': 'http://api.toopher.test/v1/pairings/{0}/reset?reset_authorization=abcde'.format(self.id)
+                })
+            )
+        })
+        reset_link = pairing.get_reset_link(self.api)
+        self.assertEqual('http://api.toopher.test/v1/pairings/{0}/reset?reset_authorization=abcde'.format(self.id), reset_link)
+
+    def test_email_reset_link(self):
+        response = {'id':self.id,
+                    'enabled': False,
+                    'pending': True,
+                    'user': self.user }
+        pairing = toopher.Pairing(response)
+        self.api.advanced.raw.client = HttpClientMock({
+            'pairings/{0}/send_reset_link'.format(self.id): (201,
+                                           '[]'
+            )
+        })
+        try:
+            pairing.email_reset_link(self.api, 'email')
+        except:
+            self.assertTrue(False)
```

Comment:
I was thinking the `try/except` bit made it clearer in the tests, as documentation, what behavior we were expecting. Especially with the `im.verify()` part (it raises an error if `im` isn't a valid image, not necessarily that something went wrong but that is it's behavior rather than returning True or False). I see other parts in the test with a `try/except` bit that do use `self.fail()`. Is that okay or would you prefer to get rid of the `try/except` totally?